app/components/sign-up/form-input.js

```diff
@@ -0,0 +1,28 @@
+import Component from '@glimmer/component';
+
+export default class FormInputComponent extends Component {
+  get disabledValue() {
+    const { firstName, lastName, userName, what } = this.args;
+
+    if (what === 'first-name' && firstName === '') return true;
+    if (what === 'last-name' && lastName === '') return true;
+    if (what === 'user-name' && userName === '') return true;
```

Comment:
I would suggest moving this to a tracked object and updating values accordingly
Helps readability and reduce scope of error in this case
You can refer [this](https://guides.emberjs.com/release/upgrading/current-edition/tracked-properties/)

Reply:
075a915